utilities/basic-utils/src/main/kotlin/NativeMemoryAllocator.kt

```diff
@@ -38,9 +64,12 @@ class NativeMemoryAllocator {
     private val smallChunks = LongArray(ChunkBucketSize)
     private val mediumChunks = LongArray(ChunkBucketSize)
     private val bigChunks = LongArray(ChunkBucketSize)
+    private val lock = Any()
```

Comment:
Note: using lock when compilation is actually parallel might have much bigger overhead, e.g. due to biased locking in JVM.